src/query/functions/temporal/aggregation.go

```diff
@@ -78,14 +82,52 @@ func (a aggProcessor) Init(op baseOp, controller *transform.Controller, opts tra
 	}
 }
 
+// NewQuantileOp create a new base temporal transform for quantile_over_time func.
+func NewQuantileOp(args []interface{}, optype string) (transform.Params, error) {
+	if optype != QuantileType {
+		return nil, fmt.Errorf("unknown aggregation type: %s", optype)
+	}
+
+	if len(args) != 2 {
+		return emptyOp, fmt.Errorf("invalid number of args for %s: %d", QuantileType, len(args))
+	}
+
+	q, ok := args[0].(float64)
+	if !ok {
+		return emptyOp, fmt.Errorf("unable to cast to quantile argument: %v for %s", args[0], QuantileType)
+	}
+
+	duration, ok := args[1].(time.Duration)
+	if !ok {
+		return emptyOp, fmt.Errorf("unable to cast to scalar argument: %v for %s", args[1], QuantileType)
```

Comment:
Can we push all of these parsing functions up into `promql/types.go`, similar to how [this does it](https://github.com/m3db/m3/blob/master/src/query/parser/promql/types.go#L97)?

A small gotcha is that any numeric expression is valid in Prom, i.e. `quantile_over_time( (1*2)/3, up[5m] )` is a valid query but would fail here and in many of the other temporal functions.

Reply:
Really don't wanna change this anymore. Also this query works in ours:

```
quantile_over_time((1*2)/3, service_writeTaggedBatchRaw_latency{instance=~"$instance",quantile="0.99"}[1m])               
```